x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -866,3 +876,32 @@ export function resolveMaxTimeInterval(timeIntervals: string[]): number | undefi
 
   return Number.isFinite(result) ? result : undefined;
 }
+
+export const getFiltersForDSLQuery = (
+  datafeedQuery: QueryDslQueryContainer,
+  dataViewId: string | undefined,
+  alias?: string,
+  store = FilterStateStore.APP_STATE
+): Filter[] => {
+  const defaultEmptyQuery = { bool: { must: [{ match_all: {} }] } };
+  if (isEqual(datafeedQuery, defaultEmptyQuery) === true) {
```

Comment:
I've simplified this check, the original was
```ts
  if (
     datafeedQuery &&
     !isPopulatedObject(datafeedQuery, ['match_all']) &&
     !isEqual(datafeedQuery, defaultEmptyQuery) &&
     dataViewId !== null
   ) {
```

I don't think checking for an empty `match_all` is needed as the comparison to `defaultEmptyQuery` does that.
Also we need to allow an empty `match_all` as it may be accompanied by additional bool filters.

I've removed the check for a `null` `dataViewId`. If the `dataViewId` is null, it means the data view for the job is missing, this is bad and it shouldn't be up to this function to check that.
I've moved the `null` check outside of all the places `getFiltersForDSLQuery` is called.

I've had to make `store` overridable because the dashboard locator's `getUrl` has some strange logic where only global filters are allowed in the url.

Reply:
iirc the `!isPopulatedObject(datafeedQuery, ['match_all'])` check is for when the datafeed query is like below, which is valid when created via the API:
```
  "query": {
    "match_all": {}
  },
```
Although in the UI, I don't imagine us ever use that ^ in our wizards/job creation process.

Reply:
ok, it will probably be safer to just have an explicit check for that shaped query. Similar to the `defaultEmptyQuery` check

Reply:
I've added a check for some known queries as part of this check. https://github.com/elastic/kibana/pull/142139/commits/a468dbbc744849c85aa753beb61da23e120b778f
I don't think it's worth being too clever with this check, if in doubt, we'll just add the filter.